From 210cc2a7c8564fdcd572fcb627054853708eee66 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Thu, 15 Nov 2018 19:17:20 -0700
Subject: [PATCH 5/8] klte-common: Move change of ownership of ES705 UART later

* Samsung loads the firmware through the front door. Pie init
  runs faster than previously, so we need to wait to change
  the ownership until some point later, after the firmware has
  time to load, else the load gets blocked by a dac_override.

Change-Id: I37be9e22b49735388ce1f5b922132dbfd20e2dc3
---
 rootdir/etc/init.qcom.rc | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/rootdir/etc/init.qcom.rc b/rootdir/etc/init.qcom.rc
index 7f508b7..58ff196 100644
--- a/rootdir/etc/init.qcom.rc
+++ b/rootdir/etc/init.qcom.rc
@@ -279,14 +279,6 @@ on boot
     chown radio system /efs/bluetooth
     chmod 0775 /efs/bluetooth
 
-    # Audience ES705 UART
-    # Note: DO NOT move this to ueventd.qcom.rc. Samsung thoughtfully has
-    #       the kernel write directly to /dev/ttyHS3 (!) to load the audience
-    #       firmware. Setting ownership immediately with ueventd would require
-    #       allowing the kernel dac_override, which is an sepolicy neverallow.
-    chmod 0660 /dev/ttyHS3
-    chown media audio /dev/ttyHS3
-
     #Create QMUX deamon socket area
     mkdir /dev/socket/qmux_radio 0770 radio radio
     chmod 2770 /dev/socket/qmux_radio
@@ -937,6 +929,14 @@ on property:wlan.driver.status=ok
     start macloader
 
 on property:sys.boot_completed=1
+    # Audience ES705 UART
+    # Note: DO NOT move this to ueventd.qcom.rc. Samsung thoughtfully has
+    #       the kernel write directly to /dev/ttyHS3 (!) to load the audience
+    #       firmware. Setting ownership immediately with ueventd would require
+    #       allowing the kernel dac_override, which is an sepolicy neverallow.
+    chmod 0660 /dev/ttyHS3
+    chown media audio /dev/ttyHS3
+
     setprop sys.io.scheduler bfq
 
     # Configure the hardware properly
-- 
2.17.1

